hbase-server/src/main/java/org/apache/hadoop/hbase/master/WorkerAssigner.java

```diff
@@ -60,27 +59,23 @@ public synchronized Optional<ServerName> acquire() {
       .findAny();
     worker.ifPresent(name -> currentWorkers.compute(name, (serverName,
       availableWorker) -> availableWorker == null ? maxTasks - 1 : availableWorker - 1));
+    event.suspend();
```

Comment:
Should only suspend the procedure when there is no worker available? And I think here we could make the return value as ServerName instead of Optional<ServerName>, and throw ProcedureSuspendedException directly if worker is not available.

Reply:
Ok. Let me try. Thanks Duo.